### DISCUSSION TITLE
Can you please add inpainting with FLUX DEV?

### DISCUSSION THREAD

**Discussion by marinathegreat** [photopea/photopea#7404] [Ideas]
Hi,
The new flux fill dev model is far superior to the current "magic replace" function inside photopea.
you can see the model in action here: https://replicate.com/black-forest-labs/flux-fill-dev
We're a paying user and we'd pay good money to be able to use FLUX dev fill inside photopea.
Thanks!

**Comment by photopea** [comment]
That looks nice! Do you know about any company which provides this service through some API, for a reasonable cost?

Right now, we pay $0.003 for each usage of the Magic Replace, and we allow up to 1000 usages per month to Premium users. I guess we would have to charge more AI credits for using Flux.

BTW. what about Stable Diffusion XL? is it better than Flux?

**Reply by marinathegreat** [reply]
All these providers allow using it commercially on their API:
replicate.com , fal.ai , mystic.ai and together.ai
It is more expensive but its worth it cause it usually gets the inpainting prompts right on the first try! 
Maybe we can have an option to switch between the 2 models? flux for more demanding tasks and the current magic replace for less complicated tasks.
AFAIK, Flux is the most capable commercially available model at the moment. It follows prompts better than SDXL.

**Reply by photopea** [reply]
It seems like a lot of work to integrate these APIs, and probably not that many people need them.

We could probably offer 10 usages of the current inpainting for the price of one FLUX inpainting.

**Reply by marinathegreat** [reply]
I can't speak for your entire user base, but from a business user perspective - I'd actually prefer paying 10x more for something that works first try. It's much easier to tell my team "do this once and move on" vs "keep trying until it works right." When creating SOPs, reliability beats cost-per-use. Maybe you could offer both options and let users choose?

**Reply by photopea** [reply]
Only 28% of the usage of Magic Replace comes from Premium users. Other users must watch two video ads each time they use it. I don't think they would like to watch 20 ads each time they use it (as Flux is 10x more expensive). So it is not worth implementing for such a tiny fraction of users.

The purpose of Photopea Premium is that you pay for using our editor. The purpose of Photopea Premium is not to resell services of others, and give you the editor for free.

Maybe if there are more premium users (e.g. at least 0.1% of all Photopea users), and Flux costs a reasonable price, we can do it. But you are asking me to pay hunreds of USD each month for renting some GPU without knowing if it will be worth it.

**Reply by xyzzie** [reply]
oh,i see that thread now..
